crypto/fipsmodule/rand/asm/rndr-armv8.pl

```diff
@@ -0,0 +1,75 @@
+#! /usr/bin/env perl
+
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+# RNDR from ARMv8.5-A.
+# System register encoding: s3_3_c2_c4_0
+# see https://developer.arm.com/documentation/ddi0601/2024-09/AArch64-Registers/RNDR--Random-Number
+
+# The first two arguments should always be the flavour and output file path.
+if ($#ARGV < 1) { die "Not enough arguments provided.
+  Two arguments are necessary: the flavour and the output file path."; }
+
+my $flavour = shift;
+my $output  = shift;
+
+$0 =~ m/(.*[\/\\])[^\/\\]+$/; $dir=$1;
+( $xlate="${dir}arm-xlate.pl" and -f $xlate ) or
+( $xlate="${dir}../../../perlasm/arm-xlate.pl" and -f $xlate) or
+die "can't locate arm-xlate.pl";
+
+open OUT,"| \"$^X\" $xlate $flavour $output";
+*STDOUT=*OUT;
+
+$code.=<<___;
+#include <openssl/arm_arch.h>
+
+.arch armv8-a
+.text
+
+# int CRYPTO_rndr(uint8_t *out, const size_t len)
+.globl CRYPTO_rndr
+.type CRYPTO_rndr,%function
+.align 4
+CRYPTO_rndr:
+  cbz x1, .Lrndr_error        // len = 0 is not supported
+  mov x4, x1                  // len: requested number of bytes
+  mov x2, #0                  // Counts number of bytes generated
+
+.Lrndr_loop:
+  mrs x3, s3_3_c2_c4_0        // rndr instruction
+  cbz x3, .Lrndr_error        // Check if RNDR failed
+
+  cmp x1, #8                  // Sets N if strictly less than 8 bytes left
+  blt .Lrndr_less_than_8_bytes
+
+  str x3, [x0], #8            // Copy 8 bytes to *out and increment pointer by 8
+  add x2, x2, #8              // Adds 8 to counter
+  sub x1, x1, #8
+  cbz x1, .Lrndr_done         // If multiple of 8 this will be 0 eventually
+  b .Lrndr_loop
+
+.Lrndr_less_than_8_bytes:
```

Comment:
I think it'd be easier if you handled this part in C. In other words, implement `CRYPTO_rndr` to be a single call to `rndr` that returns 8 bytes and then use this function in C code?

Reply:
actually, I see that the x86 implementation does something similar:
- https://github.com/aws/aws-lc/blob/905e9d037561bd54be19b1aff5a6c46b23f4ec35/crypto/fipsmodule/rand/asm/rdrand-x86_64.pl#L74
- https://github.com/aws/aws-lc/blob/905e9d037561bd54be19b1aff5a6c46b23f4ec35/crypto/fipsmodule/rand/rand.c#L304

can we do the same for Arm and also reuse the C code?

Reply:
The `rdrand` is handled that way. I didn't do that on purpose to avoid having to do extra buffer allocation and copying.
Maybe easier, but the assembly code is not complicated.

Reply:
If you insist, I would rather just make it 8-byte multiple only support.

Reply:
fair enough